playbooks/infra/deploy-bm-hypervisor.yml

```diff
@@ -0,0 +1,226 @@
+        kickstart_iso_net_config: "{{ hostvars['hypervisor'].net_config | from_yaml }}"
+
+    - name: Setup http storage
+      ansible.builtin.import_role:
+        name: redhatci.ocp.setup_http_store
+
+- name: Deploy Bare-Metal
+  hosts: hypervisor
+  gather_facts: false
+  become: true
+  vars:
+    system_rpm_path: "/tmp/{{ system_rpm_link | basename }}"
+  tasks:
+    - name: Boot BM using pre-configured ISO
+      ansible.builtin.import_role:
+        name: redhatci.ocp.boot_iso
+      vars:
+        boot_iso_url: "http://{{ hostvars['bastion']['ansible_host'] }}/{{ hostvars['bastion']['iso_name'] }}"
+
+    - name: Wait until BM installation is completed
+      ansible.builtin.wait_for_connection:
+        delay: 360
+        sleep: 10
+        timeout: 7200
+      notify:
+        - Remove installation ISO
+
+    - name: Get system rpm from repository
+      ansible.builtin.get_url:
+        url: "{{ system_rpm_link }}"
+        dest: "{{ system_rpm_path }}"
+        force: false
+        mode: "0640"
+
+    - name: Install system rpm
+      ansible.builtin.dnf:
+        name: "{{ system_rpm_path }}"
+        state: present
+        disable_gpg_check: true
+
+    - name: Activate OS
+      ansible.builtin.command:
+        "{{ activate_system_cmd }}"
+      changed_when: false
+
+    - name: Set paswordless sudo
+      ansible.builtin.lineinfile:
+        path: /etc/sudoers.d/{{ ansible_user }}
+        line: "{{ ansible_user }} ALL=(ALL) NOPASSWD: ALL"
+        mode: "0640"
+        create: true
+
+    - name: Configure network connection baremetal
+      vars:
+        ipaddr_mask: "{{ (net_config | from_yaml)['ip'] }}/{{ (net_config | from_yaml)['mask'] }}"
+      community.general.nmcli:
+        type: bridge
+        conn_name: bridge-baremetal
+        method4: manual
+        method6: disabled
+        state: present
+        stp: false
+        ifname: baremetal
+        autoconnect: true
+        ip4: "{{ ipaddr_mask | ansible.utils.ipaddr('address/prefix') }}"
+        gw4: "{{ (net_config | from_yaml)['gw'] }}"
+        dns4:
+          - "{{ (net_config | from_yaml)['dns'] }}"
+
+
+    - name: Set up network connection bridge-slave
+      community.general.nmcli:
+        type: ethernet
+        slave_type: bridge
+        ifname: "{{ (net_config | from_yaml)['interface_name'] }}"
+        master: baremetal
+        method4: disabled
+        conn_name: "{{ (net_config | from_yaml)['interface_name'] }}"
+        state: present
+        autoconnect: true
+
+    - name: Reload NetworkManager connections
+      ansible.builtin.shell: |
+        "nmcli con down  {{ (net_config | from_yaml)['interface_name'] }} &&
+         nmcli con up  {{ (net_config | from_yaml)['interface_name'] }} &&
+         nmcli con up bridge-baremetal && nmcli con up  {{ (net_config | from_yaml)['interface_name'] }}"
+      changed_when: true
+
+    - name: Gather facts
+      ansible.builtin.gather_facts:
```

Comment:
Can you please add `true` here, just for clarity and readability?

Reply:
It causes linting error.

Reply:
Correct. Adding gather_facts as task like here is true implicitly. The expected value is a dict, not a boolean.